docs/dg/dev/architecture/redis-valkey-key-value-storage.md

```diff
@@ -0,0 +1,114 @@
+
+```php
+// Configuration still uses REDIS naming for historical compatibility
+$config[StorageConstants::STORAGE_REDIS_PROTOCOL] = 'tcp';
+$config[StorageConstants::STORAGE_REDIS_HOST] = '127.0.0.1';
+$config[StorageConstants::STORAGE_REDIS_PORT] = 6379;
+```
+
+This naming convention is preserved to ensure backward compatibility and avoid breaking changes in existing Spryker installations.
+
+## Benefits of Valkey
+
+### Performance Boost
+
+Valkey introduces advanced multi-threading capabilities. In our internal tests, we've observed up to 5x higher throughput for write operations with Valkey, alongside more stable latencies for other operations. This means data can be processed and served quicker, allowing your application to handle more simultaneous connections and deliver a faster experience, especially during peak traffic periods.
+
+### Enhanced Scalability & Reliability
+
+With Valkey's improved cluster failover mechanisms and superior scaling capabilities, your Spryker platform will be even better equipped to support larger workloads and complex use cases as your business grows.
+
+### Improved Efficiency
+
+Valkey is designed for better memory efficiency, which can contribute to optimized resource utilization and overall system performance.
+
+### Continued Security Patches, Simplified Licensing, and Strong Long-Term Support
+
+The version of Redis currently in use (6.2) has reached its end-of-life, meaning it no longer receives security patches. Migrating to Valkey not only restores ongoing maintenance and vital security updates but also offers a clear and simple BSD 3 license. This contrasts with the Redis's more recent complex triple licensing model (RSALv2, SSPLv1, AGPLv3). Valkey is a Linux Foundation project and enjoys broad industry backing from major tech leaders including AWS, Google Cloud, Oracle, Snap Inc., OVH, and Percona, ensuring robust long-term support and continuous innovation. This move future-proofs a critical component of your infrastructure.
+
+## Technical implementation in Spryker
+
+### Code references
+
+Throughout Spryker's codebase, you'll encounter references to "Redis" in:
+
+- Configuration constants (e.g., `StorageConstants::STORAGE_REDIS_*`)
+- Module names and class names
+- Comments and documentation
+- Method names and variable names
+
+These naming conventions are intentionally preserved for backward compatibility. Whether you're using the Redis or Valkey as your underlying key-value store, the Spryker application interfaces remain consistent.
+
+### Data structure and operations
+
+Both Redis and Valkey support the same data structures used by Spryker:
+
+- Strings for simple key-value pairs
+- Hashes for storing objects
+- Lists for ordered collections
+- Sets for unique collections
+- Sorted sets for scored collections
+
+All Redis commands used by Spryker (GET, SET, HGET, HSET, LPUSH, LPOP, SADD, ZADD, etc.) work identically with Valkey.
```

Comment:
stupid question. Is it really HGET HSET in the end or should it be MGET and MSET?